### ISSUE TITLE
ExternalSelectMenu initial_option does not automatically serialize Option object

### ISSUE THREAD

**Issue by K8sKween** [nicklambourne/slackblocks#73]
Problem:
When trying to resolve an ExternalSelectMenu inside of an Input block the initial_option accepts Option | OptionGroup, but does not serialize it to JSON before sending to slack.

Workaround:
-When creating the blocks pre-resolving the initial option via `._resolve()`

Does not work:
```
  try:
    return InputBlock(
      label="example label",
      optional=True,
      block_id="deployable_apps_tag_select",
      element=ExternalSelectMenu(
        action_id="deployable_apps_select_tag",
        placeholder="example placeholder",
        min_query_length=0,
        initial_option=Option(text=Text(text="test_text", type_=TextType.PLAINTEXT), value="test_value")
      )
    )
  except Exception as e:
    logger.warning(f"Failed to create tag select block: {e})")
    raise e
```
    
Workaround:
```
  try:
    return InputBlock(
      label="example label",
      optional=True,
      block_id="deployable_apps_tag_select",
      element=ExternalSelectMenu(
        action_id="deployable_apps_select_tag",
        placeholder="example placeholder",
        min_query_length=0,
        initial_option=Option(text=Text(text="test_text", type_=TextType.PLAINTEXT), value="test_value")._resolve()
      )
    )
  except Exception as e:
    logger.warning(f"Failed to create tag select block: {e})")
    raise e
```

Possible solution:

Updating the `_resolve()` function here to automatically resolve the initial option
https://github.com/nicklambourne/slackblocks/blob/843cc7af53cb4e224aae4c810161658b254af1e5/slackblocks/elements.py#L1272

Caveats and Questions:
-Being that this is not a `MultiExternalSelectMenu`, it may not accept multiple options as an initial value and the schema should be checked to see if the constructor for `ExternalSelectMenu` should only accept `Option` as opposed to `Option | OptionGroup`

-This issue seems to be isolated to just the ExternalSelectMenu as it is working correctly for `MultiExternalSelectMenu`
https://github.com/nicklambourne/slackblocks/blob/843cc7af53cb4e224aae4c810161658b254af1e5/slackblocks/elements.py#L640

**Comment by nicklambourne** [comment]
Thanks for the PR, much appreciated. This fix should roll out with `v1.0.5` in the next few minutes.

**Comment by K8sKween** [comment]
I'm glad I was able to contribute, this project has been a blessing for working with slack.
